### PR TITLE
Fix logging traceback to `rotated.log`

### PR DESCRIPTION
**Pull request by AugustoPeres**
This pull request ensures that the traceback of the exception is written to the `rotated.log` file.

## QA

Try to launch more machines that you have access to:

```
import inductiva

mg = inductiva.resources.ElasticMachineGroup(machine_type="c2-standard-16",
                                             max_machines=80,
                                             disk_size_gb=70,
                                             register=True)
mg.start()


# make sure that if it passes the machine group is terminated
mg.terminate()
```

In your terminal you should see the logs:

```
Registering ElasticMachineGroup configurations:
Registering machine group failed with exception (403)
Reason: Forbidden
HTTP response headers: HTTPHeaderDict({'content-type': 'application/json', 'X-Cloud-Trace-Context': 'c5437753569ab6cbd777cc4fcec416d1', 'Date': 'Mon, 29 Jan 2024 10:38:47 GMT', 'Server': 'Google Frontend', 'Content-Length': '96'})
HTTP response body: b'{"detail":"Maximum allowed number of machines is exceeded. Maximum allowed: 10. Requested: 80."}'
```

And on the rotated file you should see:

```
2024-01-29 10:40:06|INFO|absl|machines.py:160 Registering ElasticMachineGroup configurations:
2024-01-29 10:40:06|ERROR|root|log.py:17 Registering machine group failed with exception (403)
Reason: Forbidden
HTTP response headers: HTTPHeaderDict({'content-type': 'application/json', 'X-Cloud-Trace-Context': 'b168b0008aeadd8fb37024f109d189a2', 'Date': 'Mon, 29 Jan 2024 10:40:06 GMT', 'Server': 'Google Frontend', 'Content-Length': '96'})
HTTP response body: b'{"detail":"Maximum allowed number of machines is exceeded. Maximum allowed: 10. Requested: 80."}'
Traceback (most recent call last):
  File "/Users/augustoperes/Documents/inductiva/inductiva/resources/machines_base.py", line 84, in _register_machine_group
    resp = self._api.register_vm_group(body=instance_group_config)
  File "/Users/augustoperes/Documents/inductiva/inductiva/client/paths/compute_group/post.py", line 303, in register_vm_group
    return self._register_vm_group_oapg(
  File "/Users/augustoperes/Documents/inductiva/inductiva/client/paths/compute_group/post.py", line 219, in _register_vm_group_oapg
    raise exceptions.ApiException(status=response.status,
inductiva.client.exceptions.ApiException: (403)
Reason: Forbidden
HTTP response headers: HTTPHeaderDict({'content-type': 'application/json', 'X-Cloud-Trace-Context': 'b168b0008aeadd8fb37024f109d189a2', 'Date': 'Mon, 29 Jan 2024 10:40:06 GMT', 'Server': 'Google Frontend', 'Content-Length': '96'})
HTTP response body: b'{"detail":"Maximum allowed number of machines is exceeded. Maximum allowed: 10. Requested: 80."}'
```

### CODE REVIEW

**Comment by AugustoPeres** [comment]
Note: Yapf was failing in a file not changed original in the pull request so I also formatted that file.

File: inductiva/resources/machines_base.py [hunk 1]
```diff
@@ -85,8 +84,9 @@ def _register_machine_group(self, **kwargs):
             resp = self._api.register_vm_group(body=instance_group_config)
         except (exceptions.ApiValueError, exceptions.ApiException) as e:
             logs.log_and_exit(logging.getLogger(), logging.ERROR,
-                              "Resource registering failed with exception %s",
-                              e)
+                              "Registering machine group failed"\
+                              " with exception %s", e,
```
Comment (sergiomsantos) [inline inductiva/resources/machines_base.py:88]: don't we have a way of keeping this lines together?

```python
logs.log_and_exit(
        logging.getLogger(), logging.ERROR,
        "Registering machine group failed with exception %s",
        e, exc_info=e)
```

File: inductiva/resources/machines_base.py [hunk 2]
```diff
@@ -146,9 +146,11 @@ def start(self, **kwargs):
...
             self._api.start_vm_group(body=request_body)
         except inductiva.client.ApiException as e:
-            logs.log_and_exit(logging.getLogger(), logging.ERROR,
-                              "Starting machine group failed" 
-                              " with exception %s", e)
+            logs.log_and_exit(logging.getLogger(),
+                              logging.ERROR, "Starting machine group failed"
+                              " with exception %s",
+                              e,
+                              exc_info=e)
```
Comment (sergiomsantos) [inline inductiva/resources/machines_base.py:153]: same here

File: inductiva/tasks/run_simulation.py [hunk 3]
```diff
@@ -86,8 +86,9 @@ def run_simulation(
     logging.info(
         "Consider tracking the status of the task via CLI:"
         "\n\tinductiva tasks list --task-id %s", task_id)
-    logging.info("Or, tracking the logs of the task via CLI:"
-                 "\n\tinductiva logs %s", task_id)
+    logging.info(
+        "Or, tracking the logs of the task via CLI:"
+        "\n\tinductiva logs %s", task_id)
```
Comment (sergiomsantos) [inline inductiva/tasks/run_simulation.py:91]: why these changes? does yapf complain?
I don't like having all these line breaks

**Review by sergiomsantos — COMMENTED** [review]
just a few comments about formatting.

File: inductiva/tasks/run_simulation.py (hunk 3 above)
Comment (AugustoPeres) [reply inductiva/tasks/run_simulation.py:91]: Yapf was failing for this file (which I hadn't even changed). The changes are just the result of running yapf over that file and committing it

File: inductiva/resources/machines_base.py (hunk 1 above)
Comment (AugustoPeres) [reply inductiva/resources/machines_base.py]: Done

File: inductiva/resources/machines_base.py (hunk 2 above)
Comment (AugustoPeres) [reply inductiva/resources/machines_base.py]: Done

File: inductiva/tasks/run_simulation.py (hunk 3 above)
Comment (sergiomsantos) [reply inductiva/tasks/run_simulation.py:91]: Yapf is so annoying...

**Comment by IvanPombo** [comment]
# QA Analysis

Launched 400 machines with the `MachineGroup` and received the following error:
```bash
Registering MachineGroup configurations:
Registering machine group failed with exception (403)
Reason: Forbidden
HTTP response headers: HTTPHeaderDict({'content-type': 'application/json', 'X-Cloud-Trace-Context': '73822265081cf758498f5f4d49db58b5', 'Date': 'Mon, 29 Jan 2024 16:46:18 GMT', 'Server': 'Google Frontend', 'Content-Length': '98'})
HTTP response body: b'{"detail":"Maximum allowed number of machines is exceeded. Maximum allowed: 100. Requested: 400."}'
```

In `/home/ivanpombo/.inductiva` there exists the `inductiva.log` file with the full stack trace:

```bash
2024-01-29 16:40:23|INFO|absl|machines.py:221 > Maximum estimated cloud cost  of elastic machine group: 73.504 $/h.
2024-01-29 16:40:23|INFO|root|machines_base.py:143 Starting ElasticMachineGroup(name="api-pg5v9mp0yqh9b59jj3fugqaju"). This may take a few minutes.
2024-01-29 16:40:23|INFO|root|machines_base.py:145 Note that stopping this local process will not interrupt the creation of the machine group. Please wait...
2024-01-29 16:40:55|INFO|root|machines_base.py:158 Elastic Machine Group api-pg5v9mp0yqh9b59jj3fugqaju with c2-standard-16 machines successfully started in 0:00:31.
2024-01-29 16:41:11|INFO|root|machines_base.py:167 Terminating ElasticMachineGroup(name="api-pg5v9mp0yqh9b59jj3fugqaju") This may take a few minutes.
2024-01-29 16:44:24|INFO|root|machines_base.py:183 Elastic Machine Group api-pg5v9mp0yqh9b59jj3fugqaju with c2-standard-16 machines successfully terminated in 0:03:12.
2024-01-29 16:46:17|INFO|absl|machines.py:49 Registering MachineGroup configurations:
2024-01-29 16:46:18|ERROR|root|log.py:35 Registering machine group failed with exception (403)
Reason: Forbidden
HTTP response headers: HTTPHeaderDict({'content-type': 'application/json', 'X-Cloud-Trace-Context': '73822265081cf758498f5f4d49db58b5', 'Date': 'Mon, 29 Jan 2024 16:46:18 GMT', 'Server': 'Google Frontend', 'Content-Length': '98'})
HTTP response body: b'{"detail":"Maximum allowed number of machines is exceeded. Maximum allowed: 100. Requested: 400."}'
Traceback (most recent call last):
  File "/Users/ivanpombo/Documents/GitHub/inductiva/inductiva/resources/machines_base.py", line 84, in _register_machine_group
    resp = self._api.register_vm_group(body=instance_group_config)
  File "/Users/ivanpombo/Documents/GitHub/inductiva/inductiva/client/paths/compute_group/post.py", line 303, in register_vm_group
    return self._register_vm_group_oapg(
  File "/Users/ivanpombo/Documents/GitHub/inductiva/inductiva/client/paths/compute_group/post.py", line 219, in _register_vm_group_oapg
    raise exceptions.ApiException(status=response.status,
inductiva.client.exceptions.ApiException: (403)
Reason: Forbidden
HTTP response headers: HTTPHeaderDict({'content-type': 'application/json', 'X-Cloud-Trace-Context': '73822265081cf758498f5f4d49db58b5', 'Date': 'Mon, 29 Jan 2024 16:46:18 GMT', 'Server': 'Google Frontend', 'Content-Length': '98'})
HTTP response body: b'{"detail":"Maximum allowed number of machines is exceeded. Maximum allowed: 100. Requested: 400."}'
```

Looks good to go!

**Review by sergiomsantos — APPROVED** [review]
LGTM
